### PR TITLE
Feature article structure improvement

### PR DESCRIPTION
**Pull request by gnott**
In reference to issue https://github.com/elifesciences/issues/issues/4539, a more precise detection of the ArticleXML file if a similar looking file is part of the article files. I added in lots of linting as well, and some tests to cover other edge cases.

### CODE REVIEW

File: tests/provider/test_article_structure.py [hunk 1]
```diff
@@ -79,6 +80,7 @@ def test_get_version_from_zip_filename(self, input, expected):
         {'input': 'elife-00013-vor-v1-20121015000000.zip', 'expected': 'ArticleZip'},
         {'input': 'elife-00666-v1.pdf', 'expected': 'Other'},
         {'input': 'elife-00666-v1.xml', 'expected': 'ArticleXML'},
+        {'input': 'elife-00666-supp99.xml', 'expected': 'Other'},
```
Comment (giorgiosironi) [inline tests/provider/test_article_structure.py]: this doesn't have the `-v1` suffix like the other cases, intentional?

Comment (gnott) [reply tests/provider/test_article_structure.py]: It was the name we observed in the bucket. I could also add `elife-00666-supp99-v1.xml` to the list, which should pass, and improve the test coverage.
